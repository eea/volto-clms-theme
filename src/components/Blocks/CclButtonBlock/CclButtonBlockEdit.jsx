import React from 'react';
import CclButton from '@eea/volto-clms-theme/components/CclButton/CclButton';
import { SidebarPortal } from '@plone/volto/components';
import InlineForm from '@plone/volto/components/manage/Form/InlineForm';
import { cclButtonSchema } from './Schema';
import './styles.less';

const CclButtonBlockEdit = (props) => {
  const { block, data, onChangeBlock, selected } = props;

  return (
    <>
      <div className="ccl-block-editor-header">
        <legend
          onClick={() => {
            props.setSidebarTab(1);
          }}
          aria-hidden="true"
        >
          {data.style || 'Default'} button {data.disabled && 'disabled'}
        </legend>
      </div>

      <CclButton
        url="#"
        disabled={data.disabled}
        download={data.download}
        mode={data.style}
      >
        {data.title || 'Text example...'}
      </CclButton>

      <SidebarPortal selected={selected}>
        <InlineForm
          schema={cclButtonSchema(
            Array.isArray(data?.href) && data?.href.length
              ? ['download', 'target']
              : []
          )}
          title="Button component block"
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          formData={data}
        />
      </SidebarPortal>
    </>
  );
};

export default CclButtonBlockEdit;
