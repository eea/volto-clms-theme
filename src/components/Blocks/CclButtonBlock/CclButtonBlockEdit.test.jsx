import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import CclButtonBlockEdit from './CclButtonBlockEdit';

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

const mockStore = configureStore();
test('renders an CclButtonBlockEditView hero block component', () => {
  const store = mockStore({
    content: {
      create: {},
      data: {},
    },
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const data = {
    href: 'https://www.google.com',
    disabled: false,
    style: 'default',
    download: false,
  };
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <CclButtonBlockEdit
          data={data}
          selected={false}
          block="1234"
          content={{}}
          request={{
            loading: false,
            loaded: false,
          }}
          pathname="/news"
          onChangeBlock={() => {}}
          onSelectBlock={() => {}}
          onDeleteBlock={() => {}}
          createContent={() => {}}
          onFocusPreviousBlock={() => {}}
          onFocusNextBlock={() => {}}
          handleKeyDown={() => {}}
          index={1}
        />
      </MemoryRouter>
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
