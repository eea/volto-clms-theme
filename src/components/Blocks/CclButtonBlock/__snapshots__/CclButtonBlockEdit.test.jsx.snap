// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an CclButtonBlockEditView hero block component 1`] = `
Array [
  <div
    className="ccl-block-editor-header"
  >
    <legend
      aria-hidden="true"
      onClick={[Function]}
    >
      default
       button 
    </legend>
  </div>,
  <a
    className="ccl-button ccl-button--default"
    disabled={false}
    download={false}
    href="/"
    onClick={[Function]}
  >
    Text example...
  </a>,
]
`;
